docs_samples/01_python_app_insights_overview.md

```diff
@@ -0,0 +1,239 @@
+  - Open the `.venv\Lib\site-packages` folder and delete the `psutil` and `psutil-5.9.1.dist-info` folders
+  - Then run the following:
+
+    ```Python
+    python -m pip install --upgrade pip
+    python -m pip install psutil
+    ```
+
+- Switch to the Azure Portal, navigate to your Application Insights resource
+- Under **Monitoring**, select **Logs**
+- Run the following kusto query:
+
+```kusto
+traces
+| sort by timestamp
+| where cloud_RoleName == "app.py"
+```
+
+- You should see the following:
+
+  ![The query is displayed with one result from the above program.](./media/python_simple_app_trace.png "Review the results of the query.")
+
+## Sending trace data
+
+- Open the `./SimpleApps/trace.py` file
+- Press **F5** to run the file
+- Switch to the Azure Portal
+- Browse to your lab resource group
+- Browse to the `python-appinsights-SUFFIX` application insights resource and select it
+- Under **Monitoring**, select **Logs**
+- Run the following Kusto query to see the `traces` sent to application insights:
+
+```kql
+traces
+| sort by timestamp
+| where cloud_RoleName == "trace.py"
+```
+
+- You should see your trace in the **message** column:
+
+  ![The query is displayed with one result from the above program.](./media/python_simple_trace_trace.png "Review the results of the query.")
+
+- For your trace item notice that some of the columns are empty, but others (such as `cloud_RoleName`, `cloud_RoleInstance` and `client_*`) are populated based on the client information from the SDK.
+
+## Logger methods
+
+When working with the [Python logging framework](https://docs.python.org/3/library/logging.html), you should take note of the following methods:
+
+- addHandler
+- warning
+- exception
+- setLevel
+- info
+- debug
+- critical
+
+## Capture exceptions
+
+- Switch back to your Visual Studio Code window
+- Select the `./SimpleApps/exception.py` file
+- Press **F5** to run the file, select **Python file** in the debug configuration window
+- Switch to the Azure Portal
+- Browse to your lab resource group
+- Browse to the `python-appinsights-SUFFIX` application insights resource and select it
+- Under **Monitoring**, select **Logs**
+- Run the following Kusto query to see the `exception` sent to application insights:
+
+```kql
+exceptions
+| sort by timestamp
+| where cloud_RoleName == "exception.py"
+```
+
+- Expand your exception item, notice the `details` column has the exception details you would expect, but also expand the `customDimensions` column to review the custom data that was sent to the application insights table.
+
+  ![The query is displayed with one result from the above program.](./media/python_simple_exception_custom.png "Review the results of the query.")
+
+## Utilize custom events and metrics
+
+- Switch to Visual Studio Code
+- Open a terminal window, run the following commands, be sure to replace the `SUFFIX`:
+
+```powershell
+cd Functions
+
+az login
+
+func azure functionapp publish python-appinsights-SUFFIX   
```

Comment:
Probably would be better to separate out function app examples into its own file `04_python_app_functions.md` and show examples of how to use the functions [extension](https://github.com/census-ecosystem/opencensus-python-extensions-azure/tree/main/extensions/functions#opencensus-python-azure-functions-extension)